src/main/java/io/anserini/search/queryencoder/SpladePpEdMsmarcoPassageEncoder.java

```diff
@@ -0,0 +1,161 @@
+                .addFromTextFile(getVocabPath())
+                .optUnknownToken("[UNK]")
+                .build();
+        this.tokenizer = new BertFullTokenizer(vocab, true);
+        this.environment = OrtEnvironment.getEnvironment();
+        this.session = environment.createSession(getModelPath().toString(), new OrtSession.SessionOptions());
+        System.out.println("Model loaded.");
+    }
+
+    @Override
+    public String encode(String query) throws OrtException {
+        String encodedQuery = "";
+        List<String> queryTokens = new ArrayList<>();
+        queryTokens.add("[CLS]");
+        queryTokens.addAll(tokenizer.tokenize(query));
+        queryTokens.add("[SEP]");
+
+        Map<String, OnnxTensor> inputs = new HashMap<>();
+        long[] queryTokenIds = convertTokensToIds(tokenizer, queryTokens);
+        long [][] inputTokenIds = new long[1][queryTokenIds.length];
+
+        inputTokenIds[0] = queryTokenIds;
+        long[][] attentionMask = new long[1][queryTokenIds.length];
+        long[][] tokenTypeIds = new long[1][queryTokenIds.length];
+
+        // initialize attention mask with all 1s
+        Arrays.fill(attentionMask[0], 1);
+
+        inputs.put("input_ids", OnnxTensor.createTensor(environment, inputTokenIds));
+        inputs.put("token_type_ids", OnnxTensor.createTensor(environment, tokenTypeIds));
+        inputs.put("attention_mask", OnnxTensor.createTensor(environment, attentionMask));
+
+        try (OrtSession.Result results = session.run(inputs)) {
+            float[] computedWeights = flatten(results.get("q_rep").get().getValue());
+            LinkedHashMap<String, Float> tokenWeightMap = getTokenWeightMap(computedWeights);
+            encodedQuery = generateEncodedQuery(tokenWeightMap);
+        }
+        return encodedQuery;
+    }
+
+    private long[] convertTokensToIds(BertFullTokenizer tokenizer, List<String> tokens) {
+        int numTokens = tokens.size();
+        long[] tokenIds = new long[numTokens];
+        for(int i = 0; i < numTokens; ++i) {
+            tokenIds[i] = vocab.getIndex(tokens.get(i));
+        }
+        return tokenIds;
+    }
+
+    private float[] flatten(Object obj) {
+        /* 
+         * This function flattens the list of lists into a single list of floats.
+        */
+        List<Float> weightsList = new ArrayList<>();
+        Object[] inputs = (Object[]) obj;
+        for (Object input : inputs) {
+            float[] weights = (float[]) input;
+            for (float weight : weights) {
+                weightsList.add(weight);
+            }
+        }
+        return toArray(weightsList);
+    }
+
+    private float[] toArray(List<Float> input) {
+        float[] output = new float[input.size()];
+        for (int i = 0; i < output.length; i++) {
+            output[i] = input.get(i);
+        }
+        return output;
+    }
+    private LinkedHashMap<String, Float> getTokenWeightMap(float[] computedWeights){
```

Comment:
Do we need to specify implementation in the return sig? Why not just `Map<String, Float>`?